test/distributed/elastic/rendezvous/c10d_rendezvous_backend_test.py

```diff
@@ -97,26 +168,56 @@ def test_create_backend_returns_backend_if_is_host_is_not_specified_and_store_al
 
         del self._params.config["is_host"]
 
-        self.test_create_backend_returns_backend()
+        self._assert_create_backend_returns_backend()
 
     def test_create_backend_returns_backend_if_endpoint_port_is_not_specified(self) -> None:
         self._params.endpoint = self._expected_endpoint_host
 
         self._expected_endpoint_port = 29400
 
-        self.test_create_backend_returns_backend()
+        self._assert_create_backend_returns_backend()
 
-    def test_create_backend_returns_backend_if_store_type_is_not_specified(self) -> None:
+    def test_create_backend_returns_backend_if_endpoint_file_is_not_specified(self) -> None:
+        self._params_filestore.endpoint = ""
+
+        self._run_test_with_store("file", self._assert_create_backend_returns_backend)
+
+    def create_backend_returns_backend_if_store_type_is_not_specified(self) -> None:
         del self._params.config["store_type"]
 
-        self.test_create_backend_returns_backend()
+        self._expected_store_type = TCPStore
+        if (not self._params.get("read_timeout")):
+            self._expected_read_timeout = timedelta(seconds=60)
+
+        self._assert_create_backend_returns_backend()
+
+    def test_create_backend_returns_backend_if_store_type_is_not_specified(self) -> None:
+        # If the endpoint is not cleared in the case of filestore, the tcp endpoint
+        # parser interprets the filename as a host, and returns an error.
+        # This is tested in a different test:
+        # test_create_backend_raises_error_if_store_type_not_specified_with_file_endpoint
+        self._params_filestore.endpoint = ""
+
+        for store_type in ["tcp", "file"]:
```

Comment:
I think this test is superfluous. In the original test (lines from 185 to 192) we explicitly delete the `store_type` config before assertion. Not sure if we gain any test coverage by testing it with both store types here.